src/acquisition/covid_hosp/update.py

```diff
@@ -0,0 +1,176 @@
+"""
+Acquires the "COVID-19 Reported Patient Impact and Hospital Capacity by State
+Timeseries" dataset provided by the US Department of Health & Human Services
+via healthdata.gov.
+"""
+
+# standard library
+import json
+import re
+
+# first party
+from delphi.epidata.acquisition.covid_hosp.database import Database
+from delphi.epidata.acquisition.covid_hosp.network import Network
+
+
+class UpdateException(Exception):
+  """Exception raised exclusively by the Update class."""
+
+
+class Update:
+
+  # regex to extract issue date from revision field
+  # example revision: "Mon, 11/16/2020 - 00:55"
+  REVISION_PATTERN = re.compile(r'^.*\s(\d+)/(\d+)/(\d+)\s.*$')
+
+  def get_entry(obj, *path):
+    """Get a deeply nested field from an arbitrary object.
+
+    Parameters
+    ----------
+    obj : dict
+      The object to traverse.
+    path : tuple of names and indices
+      Path to the desired field in the object.
+
+    Returns
+    -------
+    object
+      The nested object.
+
+    Raises
+    ------
+    UpdateException
+      If the field can't be found.
+    """
+
+    try:
+      for elem in path:
+        is_index = isinstance(elem, int)
+        is_list = isinstance(obj, list)
+        if is_index != is_list:
+          raise UpdateException('index given for non-list or vice versa')
+        obj = obj[elem]
+      return obj
+    except Exception as ex:
+      path_str = '/'.join(map(str, path))
+      msg = f'unable to access object path "/{path_str}"'
+      raise UpdateException(msg) from ex
+
+  def get_issue_from_revision(revision):
+    """Extract and return an issue from a revision string.
+
+    Parameters
+    ----------
+    revision : str
+      The free-form revision string.
+
+    Returns
+    -------
+    int
+      The issue in YYYYMMDD format.
+
+    Raises
+    ------
+    UpdateException
+      If the issue can't be extracted.
+    """
+
+    match = Update.REVISION_PATTERN.match(revision)
+    if not match:
+      raise UpdateException(f'unable to extract issue from "{revision}"')
+    y, m, d = match.group(3), match.group(1), match.group(2)
+    return int(y) * 10000 + int(m) * 100 + int(d)
+
+  def get_date_as_int(date):
+    """Convert a YYYY-MM-DD date from a string to a YYYYMMDD int.
```

Comment:
I notice this dataset uses multiple date formats (MM/DD/YYYY and YYYY-MM-DD). How stable do we think those formats are, and would it be wise to use something like `dateutil`?

Reply:
`YYYY-MM-DD` should be stable as that comes from the dataset, which had better have a stable format.

`MM/DD/YYYY` comes from a field in the metadata which is not described. I treat it as a free-form text field, but in practice I would bet that the format won't change.

I'm not familiar with `dateutil`. It could be wise, but I recommend deferring that upgrade until later since it's sortof speculative at this point.